src/rt-app.c

```diff
@@ -759,6 +772,15 @@ void *thread_body(void *arg)
 				NULL);
 	}
 
+	/* TODO find a better way to handle that constraint:
+	 * Set the task to SCHED_DEADLINE as far as possible touching its
+	 * budget as little as possible for the first iteration.
+	 */
+
+	/* Set scheduling policy and print pretty info on stdout */
+	log_notice("[%d] Starting with %s policy with priority %d", data->ind, policy_to_string(data->sched_data->policy), data->sched_data->prio);
+	set_thread_priority(data, data->sched_data);
```

Comment:
Yes we will call it twice in this case. We have 2 use cases that have problem:
1st config:
```
"threadA" : {
    "policy" : "SCHED_OTHER",
    "phases" : {
        "phase1" : {
            "run" : 5000,
            "timer" :  { "ref" : "tickB", "period" : 7777, }
        },
    },
}
```
The scheduling policy parameters are only at thread level but not at phase so it must be set before starting the loop. So we can't remove this call to set_thread_priority 

2nd config:
"threadA" : {
    "policy" : "SCHED_OTHER",
    "run" : 5000,
    "timer" :  { "ref" : "tickB", "period" : 7777, }
}
```
Because we don't have phase object, the scheduling policy parameters will be set at thread level and at phase level. the scheduling policy will be set twice.
But in fact, we don't have to allocate and set thread_data->sched_data in this case so the call to set_thread_priority above will return because sched_data is null. The change should be done in 
parse_thread_data() and data->sched_data = parse_sched_data(obj, opts->policy); should be called only if no "phases" object is defined. But in phase, we don't use default scheduling policy because it can be the same as previous phase so the config below will not work

```
"threadA" : {
    "run" : 5000,
    "timer" :  { "ref" : "tickB", "period" : 7777, }
}
```
I don't have simple solution for now. I will think a bit more about this